### DISCUSSION TITLE
Filebeat Help - Urgent

### DISCUSSION THREAD

**Discussion by jclewis7682** [Security-Onion-Solutions/securityonion#9165] [General]
Hello,

Looking to setup netflow with filebeat/logstash. I just created a single sec onion instance. I watched this video https://www.youtube.com/watch?v=ew5gtVjAs7g but at the 10:40 mark they talk about docker exec -i so-filebeat filebeat setup modules -pipelines -modules netflow -c /usr/share/filebeat/module-setup.yml.  However, I don't have the /usr/share/filebeat/module-setup.yml folder.  I did so-filebeat-start and setup.  However, its still not there.  What am I missing?  I searched other posts but couldnt find the answer. Any help would be greatly appreciated.  Thank you.

**Comment by TotieBash** [comment]
Honestly I think you can skip that particular section.  I think the latest SO that particular module is already loaded by default. I'm using SO 2.3.181. Check "so-filebeat-module-setup".  I use that same youtube link before as reference to setup filebeat cisco.ios module and it is still overall a very good reference.  Use the following command for troubleshooting:
1.  Check that filebeat docker container is listening on port 2055:
docker ps | grep filebeat

2. Check the docker host is listening on port 2055:
ss -tulpn  | grep 2055

3. Check iptables is open for 2055
iptables --list -n | grep 2055 

4.  Finally double check if there are reallly netflow traffic coming on the wire. 
tcpdump -i SO-MGMT-INT port 2055 -nn
example:  tcpdump -i eth1 -nn port 2055

if you dont see anything but you are confident that there should be sending then it might be sending on a different port.  Adjust the tcpdump filter to the following:  
tcpdump -i eth1 -nn host NETFLOW-SOURCE-IP && udp

**Reply by jclewis7682** [reply]
Thank you for the update.  1. Filebeat is up.  2. It is running bust instead of Listen it says UNCONN.  3.  Iptables has all of the established ranges.  4. I did not receive any tcpdumps of files.  For the last command, we are receiving syslogs (which we setup with pfsense) but not filebeat.

For netflow data, do we need to point the firewall, router, and/or switches at our Sec Onion?  Also, do we need to setup filebeat-* index or does that fall under so-*?

Thanks again for the help.  We are helping a customer on an incident.

**Reply by TotieBash** [reply]
"For netflow data, do we need to point the firewall, router, and/or switches at our Sec Onion?"  
Yes, you need to setup the network devices to point to SO management interface.  This is key.  The video quickly mention this at the first 30 seconds intro stating to the effect of if you don't have a forward node then you can set this up.  Matt Gracie is stating that because by default if all the traffic you are interested is coming through the tap interface then you do not need netflow because Zeek (or Suricata) is already ingesting those 5tuple metadata logs.

Anyways for netflow, yes you do need to configure your devices and you should verify it with tcpdump to make sure they are coming in.  Here is a truncated sample for Cisco device.
.....Cisco config truncated.....
ip flow-export destination SO-MGMT-IP 2055

**Reply by jclewis7682** [reply]
Just wanted to say thank you for the update.  This worked like a charm.  Thank you.
